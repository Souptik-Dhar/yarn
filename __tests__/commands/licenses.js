/* @flow */

import {JSONReporter} from '../../src/reporters/index.js';
import {run as buildRun} from './_helpers.js';
import {run as licenses} from '../../src/cli/commands/licenses.js';

const path = require('path');

const fixturesLoc = path.join(__dirname, '..', 'fixtures', 'licenses');

const runLicenses = buildRun.bind(
  null,
  JSONReporter,
  fixturesLoc,
  async (args, flags, config, reporter, lockfile, getStdout): Promise<string> => {
    await licenses(config, reporter, flags, args);
    return getStdout();
  },
);

test('lists all licenses of the dependencies with the --json argument', async (): Promise<void> => {
  await runLicenses(['list'], {json: true}, '', (config, reporter, stdout) => {
    expect(stdout).toMatchSnapshot();
  });
});

test('should genereate disclaimer on demand', async (): Promise<void> => {
  await runLicenses(['generate-disclaimer'], {json: true}, '', (config, reporter, stdout) => {
    expect(stdout).toMatchSnapshot();
  });
});
