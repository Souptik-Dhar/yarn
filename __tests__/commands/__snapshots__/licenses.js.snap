// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lists all licenses of the dependencies with the --json argument 1`] = `
"{\\"type\\":\\"warning\\",\\"data\\":\\"package.json: No license field\\"}
{\\"type\\":\\"warning\\",\\"data\\":\\"No license field\\"}
{\\"type\\":\\"tree\\",\\"data\\":{\\"type\\":\\"licenses\\",\\"trees\\":[{\\"name\\":\\"is-plain-obj@1.1.0\\",\\"children\\":[{\\"name\\":\\"License: MIT\\"},{\\"name\\":\\"URL: https://github.com/sindresorhus/is-plain-obj.git\\"}]}]}}
"
`;

exports[`should genereate disclaimer on demand 1`] = `
"{\\"type\\":\\"warning\\",\\"data\\":\\"package.json: No license field\\"}
{\\"type\\":\\"warning\\",\\"data\\":\\"No license field\\"}
{\\"type\\":\\"warning\\",\\"data\\":\\"package.json: No license field\\"}
{\\"type\\":\\"log\\",\\"data\\":\\"THE FOLLOWING SETS FORTH ATTRIBUTION NOTICES FOR THIRD PARTY SOFTWARE THAT MAY BE CONTAINED IN PORTIONS OF THE  PRODUCT.\\\\n\\"}
{\\"type\\":\\"log\\",\\"data\\":\\"-----\\\\n\\"}
{\\"type\\":\\"log\\",\\"data\\":\\"The following software may be included in this product: is-plain-obj. A copy of the source code may be downloaded from https://github.com/sindresorhus/is-plain-obj.git. This software contains the following license and notice below:\\\\n\\"}
{\\"type\\":\\"log\\",\\"data\\":\\"The MIT License (MIT)\\\\n\\\\nCopyright (c) Sindre Sorhus <sindresorhus@gmail.com> (sindresorhus.com)\\\\n\\\\nPermission is hereby granted, free of charge, to any person obtaining a copy\\\\nof this software and associated documentation files (the \\\\\\"Software\\\\\\"), to deal\\\\nin the Software without restriction, including without limitation the rights\\\\nto use, copy, modify, merge, publish, distribute, sublicense, and/or sell\\\\ncopies of the Software, and to permit persons to whom the Software is\\\\nfurnished to do so, subject to the following conditions:\\\\n\\\\nThe above copyright notice and this permission notice shall be included in\\\\nall copies or substantial portions of the Software.\\\\n\\\\nTHE SOFTWARE IS PROVIDED \\\\\\"AS IS\\\\\\", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR\\\\nIMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,\\\\nFITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE\\\\nAUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER\\\\nLIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,\\\\nOUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN\\\\nTHE SOFTWARE.\\"}
{\\"type\\":\\"log\\",\\"data\\":\\"\\"}
"
`;
